pkg/controller/core.oam.dev/v1alpha2/application/generator.go

```diff
@@ -329,11 +329,16 @@ func (h *AppHandler) checkComponentHealth(appParser *appfile.Parser, appRev *v1b
 			dispatchResources = append([]*unstructured.Unstructured{readyWorkload}, readyTraits...)
 		}
 		if !h.resourceKeeper.ContainsResources(dispatchResources) {
-			return false, err
+			return false, nil, nil, err
 		}
 
 		_, isHealth, err := h.collectHealthStatus(auth.ContextWithUserInfo(ctx, h.app), wl, appRev, overrideNamespace, false)
-		return isHealth, err
+		if err != nil {
+			return false, nil, nil, err
+		}
+
+		workload, traits, err := getComponentResources(auth.ContextWithUserInfo(ctx, h.app), manifest, wl.SkipApplyWorkload, h.r.Client)
```

Comment:
I think this line of code will add extra "get" requests, right?

Reply:
Yes, it will get one more time for resources. Maybe I can make a optimization. Let me try.

Reply:
great!